nservicebus/tools/migrate-to-native-delivery.md

```diff
@@ -0,0 +1,233 @@
+                        [--endpoint] <endpointName>
+                        [--allendpoints]
+```
+
+NOTE: The listed endpoints will be the in the escaped form that is used to prefix the endpoints timeout table
+
+### Aborting a migration
+
+To abort an ongoing migration use the `abort` command.
+
+**RavenDB**
+
+```
+migrate-timeouts ravendb abort
+                        --serverUrl <serverUrl>
+                        --databaseName <databaseName>
+                        [--prefix] <prefix>
+                        [--ravenVersion] <ravenVersion>
+```
+
+**Sql Persistence**
+
+```
+migrate-timeouts sqlp abort
+                        --source <source>
+                        --dialect <sqlDialect>
+```
+
+## How the tool works
+
+The migration tool will perform a few health checks:
+
+- verify it's able to connect to the storage
+- verify it's able to connect to the target transport
+- verify that the target transport supports native delayed delivery
+- check that necessary infrastructure for native delays is setup for all delayed message destinations found
+- list all the endpoints for which the tool can detect timeouts
+- calculate the amount of timeouts to migrate per endpoint
+- validate if there are timeouts the tool is [unable to migrate](migrate-to-native-delivery.md#limitations)
+
+Once this information has been reviewed, the migration process can be started.
+Even though the tool supports migrating all endpoints connected to the persistence at once, it is highly suggested to migrate endpoint by endpoint, especially for critical endpoints. Even when selecting the `--allEndpoints` option, the tool will execute an endpoint-by-endpoint migration behind the scenes.
+
+## Cleanup
+
+The tool will not delete any timeouts or tables to prevent message loss. This section describes how to cleanup archived timeouts and remove no longer used tables once the migration have been verified.
+
+WARN: This is a destructive operation and should only be performed once its been verified that the migration has been successful.
+
+### RavenDB
+
+Delete all the documents in the `TimeoutDatas` collection that have an `OwningTimeoutManager` starting with `__migrated__`.
+
+### Sql persistence
+
+Use `SELECT * FROM TimeoutsMigration_State` to list all performed migrations. For all the successful once do the following:
+
+- Make sure that [RabbitMQ compatibility mode](/transports/rabbitmq/delayed-delivery.md#backwards-compatibility) is turned off
+- Delete the `{EndpointName}_TimeoutData` table
+- Delete the migration table named `TimeoutData_migration_{MigrationRunId}`, where `MigrationRunId` is taken from the output of the `TimeoutsMigration_State` query.
+
+## Limitations
+
+As documented in the [RabbitMQ transport](/transports/rabbitmq/delayed-delivery.md), the maximum delay value of a timeout is 8,5 years. If the migration tool encounters any timeouts that have delivery time set beyond that, it won't be possible to migrate that endpoint.
```

Comment:
```suggestion
As documented in the [RabbitMQ transport](/transports/rabbitmq/delayed-delivery.md), the maximum delay value of a timeout is 8 and a half years. If the migration tool encounters any timeouts that have delivery time set beyond that, it will not migrate that endpoint's timeouts.
```